modules/javafx.controls/src/main/java/javafx/scene/control/skin/VirtualFlow.java

```diff
@@ -805,6 +805,9 @@ public final BooleanProperty verticalProperty() {
         if (vertical == null) {
             vertical = new BooleanPropertyBase(true) {
                 @Override protected void invalidated() {
+                    resetIndex(cells);
```

Comment:
minor: [cells|pile]clear() can be combined with resetIndex(), something like

```
    private void clear(ArrayLinkedList<T> cells) {
        for (T cell : cells) {
            cell.updateIndex(-1);
        }
        cells.clear();
    }
```

as these operations seem to be called in pairs

Reply:
Unfortunately not in the `needsRebuildCells` if condition, where we only reset the index and add all cells to the pile

Reply:
you are right.